java/client/src/main/java/com/youtube/vitess/client/cursor/SimpleCursor.java

```diff
@@ -59,10 +59,7 @@ public boolean next() throws SQLException {
   }
 
   @Override
-  protected Row getCurrentRow() throws SQLException {
-    if (row == null) {
-      throw new SQLDataException("no current row");
-    }
+  public Row getCurrentRow() {
```

Comment:
I think getCurrentRow should be a trivial method call that returns the current row without additional operation. We throw SQLDataException in "next" method already.

If there is no current row, this really means DB returns an empty result which in my opinion is a valid state. Therefore, I feel getCurrentRow() should return null in such case. If DB does return multiple rows, then "next" method must throw a SQLDataException if there are any problems.